<a href="https://colab.research.google.com/github/Kamal-Moha/Stroke_Prediction/blob/main/Project_2_Picking_Datasets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

# **First Choice of Dataset**

**Stroke Prediction Dataset**

Source:

I got this dataset from Kaggle. Here's the link to the dataset


[Dataset source](https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset)



This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.

In [5]:
# Let's have a look at the dataset
path = '/content/drive/MyDrive/CODING DOJO DS BOOTCAMP/PROJECTS/PROJECT 2/Stroke Prediction Dataset/healthcare-dataset-stroke-data.csv'
df = pd.read_csv(path)
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

The target is the `stoke` column which has 1 if the patient had a stroke and 0 if not.

**Is this a classification or regression problem?**

This is a Binary Classification problem having only 2 options in our target vector.

1 if the patient had a stroke 0 if not.

**How many features?**

In [8]:
df.columns

Index(['id', 'gender', 'age', 'hypertension', 'heart_disease', 'ever_married',
       'work_type', 'Residence_type', 'avg_glucose_level', 'bmi',
       'smoking_status', 'stroke'],
      dtype='object')

There are 11 features that are going to affect our modeling of this dataset

**How many rows of data.**

In [7]:
df.shape

(5110, 12)

There are 5110 rows in our data

**Foreseeable Challenges**

There could be duplicates, missing values, outliers, etc in our data which could exploring & modeling challenging.

# **Second Choice of Dataset**

**Predicting Flight Fares**

Source:

I got this dataset from Kaggle. Here's the link to the dataset

[Dataset source](https://www.kaggle.com/datasets/yashdharme36/airfare-ml-predicting-flight-fares)

A dataset for predicting the fare of a flight. By analyzing patterns in flight fares over time, users can identify the best times to book tickets and potentially save money.

In [9]:
path = '/content/drive/MyDrive/CODING DOJO DS BOOTCAMP/PROJECTS/PROJECT 2/AirFare Prediction/Cleaned_dataset.csv'
df = pd.read_csv(path)
df.head()

Date_of_journey Journey_day    Airline Flight_code    Class Source  \
0      2023-01-16      Monday   SpiceJet     SG-8169  Economy  Delhi   
1      2023-01-16      Monday     Indigo     6E-2519  Economy  Delhi   
2      2023-01-16      Monday   GO FIRST      G8-354  Economy  Delhi   
3      2023-01-16      Monday   SpiceJet     SG-8709  Economy  Delhi   
4      2023-01-16      Monday  Air India      AI-805  Economy  Delhi   

    Departure Total_stops      Arrival Destination  Duration_in_hours  \
0  After 6 PM    non-stop   After 6 PM      Mumbai             2.0833   
1  After 6 PM    non-stop  Before 6 AM      Mumbai             2.3333   
2  After 6 PM    non-stop  Before 6 AM      Mumbai             2.1667   
3  After 6 PM    non-stop   After 6 PM      Mumbai             2.0833   
4  After 6 PM    non-stop   After 6 PM      Mumbai             2.1667   

   Days_left  Fare  
0          1  5335  
1          1  5899  
2          1  5801  
3          1  5794  
4          1  5955

The target is the `Fare` column which has the cost of the ticket per flight.

**Is this a classification or regression problem?**

This is a regression problem in which we shall predict the cost of a flight ticket using features like `Class`, `Days_left`, `Journey_day`, etc

**How many features?**

In [10]:
df.columns

Index(['Date_of_journey', 'Journey_day', 'Airline', 'Flight_code', 'Class',
       'Source', 'Departure', 'Total_stops', 'Arrival', 'Destination',
       'Duration_in_hours', 'Days_left', 'Fare'],
      dtype='object')

There are 12 features that are going to impact our modeling of this dataset

**How many rows of data.**

In [11]:
df.shape

(452088, 13)

There are 452088 rows of data. This is clearly a large dataset which is exciting

**Foreseeable Challenges**


This dataset does not cover variations in aviation fuel prices as this is the one of influencing factor for deciding fare.

This dataset only covers flights departing from specific airports and limited to a certain time period